src/components/Modals/CreateSingleItemModal/CreateSingleItemModal.tsx

```diff
@@ -586,15 +586,15 @@ export default class CreateSingleItemModal extends React.PureComponent<Props, St
     const { weareblePreviewUpdated, type, model } = this.state
     // if model is an image, the wearable preview won't be needed
     if (model && isImageFile(model)) {
-      this.getMetricsAndScreenshot()
+      await this.getMetricsAndScreenshot()
```

Comment:
Returning the promise works as well:
```suggestion
      return this.getMetricsAndScreenshot()
```